lib/linter/node-event-generator.js

```diff
@@ -260,9 +274,10 @@ class NodeEventGenerator {
         this.exitSelectorsByNodeType = new Map();
         this.anyTypeEnterSelectors = [];
         this.anyTypeExitSelectors = [];
+        this.knownTypes = new Set(Object.keys(esqueryOptions.visitorKeys));
```

Comment:
I was wondering if it's really necessary to keep track of *known* selector types when the goal is to detect *simple* selectors. Maybe we could just test the selector string (in `tryParseSelector` and `applySelector`) for special characters like `>`, `+`, `[`, `]`, `:`, `,`, `*`, `!`, `~` white spaces, etc. A valid selector is a simple selector if it does not contain any special characters.

Reply:
I'm not 100% sure, but I think doing such a check could be slower than keeping track of known Node types.

Reply:
I was also not sure, so I decided to test it (spoiler: it didn't help)

I defined a regex

```js
const SIMPLE_SELECTOR_REGEX = /^\w+$/;
```

It's not perfect, but it matches all standard ESTree node names plus the code path analysis events (which are intended to work like identifiers as it seems). Then I replaced `this.knownTypes.has()` with `SIMPLE_SELECTOR_REGEX.test()` [here](https://github.com/eslint/eslint/blob/99ae371bd704da6ed895c216b8063c2b01517269/lib/linter/node-event-generator.js#L197) and [here](https://github.com/eslint/eslint/blob/99ae371bd704da6ed895c216b8063c2b01517269/lib/linter/node-event-generator.js#L312) and cleared the remaining references to `knownTypes`.

I run the VSCode profiler on `npm run lint` to measure execution time and compared the results with `knownTypes` and with my regex. No difference though. The times are almost the same either way. I guess we can keep the `knownTypes` and move on.